depthai_sdk/src/depthai_sdk/visualize/visualizer.py

```diff
@@ -199,55 +200,129 @@ def serialize(self) -> str:
+                   line_width: float = None,
+                   line_height: float = None,
+                   hide_label: bool = None,
+                   label_position: TextPosition = None,
+                   label_padding: int = None) -> 'Visualizer':
         """
         Configure how bounding boxes will look like.
         Args:
-            **kwargs:
+            thickness: Thickness of the bounding box.
+            fill_transparency: Transparency of the bounding box.
+            box_roundness: Roundness of the bounding box.
+            color: Color of the bounding box.
+            bbox_style: Style of the bounding box.
+            line_width: Width of the bbox horizontal lines CORNERS or ROUNDED_CORNERS style is used.
+            line_height: Height of the bbox vertical lines when CORNERS or ROUNDED_CORNERS style is used.
+            hide_label: Flag that indicates if the label should be hidden.
+            label_position: Position of the label.
+            label_padding: Padding of the label.
 
         Returns:
-
+            self
         """
-        self.config.detection = replace(self.config.detection, **kwargs)
+        kwargs = locals()
+        kwargs.pop('self')
+        kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+        if len(kwargs) > 0:
+            self.config.detection = replace(self.config.detection, **kwargs)
+
         return self
 
-    def text(self, **kwargs: dict) -> 'Visualizer':
+    def text(self,
+             font_face: int = None,
+             font_color: Tuple[int, int, int] = None,
+             font_transparency: float = None,
+             font_scale: float = None,
+             font_thickness: int = None,
+             font_position: TextPosition = None,
+             bg_transparency: float = None,
+             bg_color: Tuple[int, int, int] = None,
+             line_type: int = None,
+             auto_scale: bool = None) -> 'Visualizer':
         """
         Configure how text will look like.
 
         Args:
-            **kwargs:
+            font_face: Font face (from cv2).
+            font_color: Font color.
+            font_transparency: Font transparency.
+            font_scale: Font scale.
+            font_thickness: Font thickness.
+            font_position: Font position.
+            bg_transparency: Text background transparency.
+            bg_color: Text background color.
+            line_type: Line type (from cv2).
+            auto_scale: Flag that indicates if the font scale should be automatically adjusted.
 
         Returns:
-
+            self
         """
-        self.config.text = replace(self.config.text, **kwargs)
+        kwargs = locals()
+        kwargs.pop('self')
+        kwargs = {k: v for k, v in kwargs.items() if v is not None}
+
+        if len(kwargs) > 0:
+            self.config.text = replace(self.config.text, **kwargs)
+
         return self
 
-    def tracking(self, **kwargs: dict) -> 'Visualizer':
+    def tracking(self,
```

Comment:
I think `max_age` would also be very useful here. Or maybe `max_lenght`, so after X pixels or % of the image it would discard older tracklets. Thoughts?

Reply:
Yeah, it's a good idea. Can be really useful for fast moving objects.

Reply:
Yeah, great idea. Added it in 907040bdd5636dfd5153eaf1d4e618ca372b4435. It's customizable via `visualizer.tracking(max_length=100)`, where the units are pixels. So here the maximum length cannot exceed 100px.